src/bridge/cockpitfsdir.c

```diff
@@ -214,19 +229,26 @@ cockpit_fsdir_prepare (CockpitChannel *channel)
                                    self->cancellable,
                                    on_enumerator_ready,
                                    self);
-  self->monitor = g_file_monitor_directory (file, 0, NULL, &error);
-  g_object_unref (file);
 
-  if (self->monitor == NULL)
+  if (!nowatch)
     {
-      g_message ("%s: %s", self->path, error->message);
-      options = cockpit_channel_close_options (channel);
-      json_object_set_string_member (options, "message", error->message);
-      problem = "internal-error";
-      goto out;
-    }
+      self->monitor = g_file_monitor_directory (file, 0, NULL, &error);
+      g_object_unref (file);
+      if (self->monitor == NULL)
+        {
+          g_message ("%s: %s", self->path, error->message);
```

Comment:
Suggest "%s: couldn't monitor directory: %s" ... otherwise this leads to hard to locate errors. We fixed a bunch of these in the cockpitfs* code already.